Add python oauth2 examples for cloud (#58)
streamnative/examples: cloud/python/oauth2/produce-consume-oauth2-client.md

# Produce and consume messages using Oauth2 authentication plugin

This example demonstrates how to use the Oauth2 authentication plugin to connect to the Pulsar service for producing and consuming messages.

## Prerequisites

- Python 3 or higher
- pulsar-client 2.6.2 or higher

## Examples

- [OAuth2 Producer](./OAuth2Producer.py)
- [OAuth2 Consumer](./OAuth2Consumer.py)

## Steps

1. Get the service URLs. For details, see [Get Pulsar service URLs](https://github.com/streamnative/pulsar-examples/tree/master/cloud#get-pulsar-service-urls).

2. Get the oauth2 authentication parameters. For details, see [Get Oauth2 authentication parameters](https://github.com/streamnative/examples/tree/master/cloud#get-oauth2-authentication-parameters).

> 3. Run a consume example to wait for the produced messages from the topic `oauth2`.

Author: ```suggestion
3. Run the Python consumer example to receive messages from the topic `oauth2`.
```

# Python produce and consume using oauth2 authentication plugin

This example demonstrates how to use oauth2 authentication plugin to connect to the pulsar service for producing and consuming.

## Prerequisites

- Python3 or higher
- pulsar-client 2.6.2 or higher

## Examples

- [OAuth2 Producer](./OAuth2Producer.py)
- [OAuth2 Consumer](./OAuth2Consumer.py)

## Steps

1. Get the service URLs. For details, see [Get Pulsar service URLs](https://github.com/streamnative/pulsar-examples/tree/master/cloud#get-pulsar-service-urls).

2. Get the oauth2 authentication parameters. For details, see [Get Oauth2 authentication parameters](https://github.com/streamnative/examples/tree/master/cloud#get-oauth2-authentication-parameters).

> 3. Run a consume example to wait for the produced messages from the topic `oauth2`.

Reviewer: 3. Run the Python consumer to receive messages from the topic `oauth2`.

    ```shell
    python3 OAuth2Consumer.py \
        -su "pulsar+ssl://streamnative.cloud:6651 \
        -t oauth2 -n 10 --auth-params '{
        "issuer_url": "https://auth.streamnative.cloud",
        "private_key": "/path/to/private.key",
        "audience": "urn:sn:pulsar:test-organization-name:test-pulsar-instance-name"}'
    ```

> 4. Open another terminal, run a producer example to produce 10 messages to the pulsar topic `oauth2`.

Reviewer: ```suggestion
4. Run the Python producer to publish messages to the topic `oauth2`.
```

    ```shell
    python3 OAuth2Producer.py \
        -su "pulsar+ssl://streamnative.cloud:6651 \
        -t oauth2 -n 10 --auth-params '{
        "issuer_url": "https://auth.streamnative.cloud",
        "private_key": "/path/to/private.key",
        "audience": "urn:sn:pulsar:test-organization-name:test-pulsar-instance-name"}'
    }'
    ```
    

>     After running this producer example, you will see the following successful message.

Reviewer: ```suggestion
    **Output**
```
